tests/unit/plugins/test_writer.py

```diff
@@ -308,6 +319,11 @@ def test_pipeline_generation(
     if framework == "DVC":
         file_names.append("dvc.yaml")
 
+        # TODO fix coverage for tests of file per task frameworks to include non artifact tasks
```

Comment:
Without if-statement the single-stage implementation fails 
![image](https://user-images.githubusercontent.com/67829287/210262574-473582ff-0491-43f8-b2ab-5e5c55fd2740.png)

Reply:
Oh this is from the single stage all sessions test right? 

I actually prefer to simply remove the test for that flavor and have better coverage for your `StagePerArtifact` flavor, 

`SingleStageAllSessions` it is much less important than this flavor and is more of a proof of concept for DVC as a whole

Reply:
Ok.

Reply:
Fixed as mentioned.